File: Titan Backend/src/main/java/com/example/demo/service/AddressService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.model.Address;

public interface AddressService 
{
   public Address saveAddress(Address address);
   public Address getAddressById(long id);
   public List<Address> getAddresses();
   public Address updateAddress(Address address);
   public void deleteAddress(long id);
}

File: Titan Frontend/src/components/AddressService.js
import axios from 'axios';

const API_BASE_URL = "http://localhost:8080"; 

class AddressService {

    saveAddress(address) {
        return axios.post(`${API_BASE_URL}/address`, address);
    }

    getAddressById(addressId) {
        return axios.get(`${API_BASE_URL}/address/${addressId}`);
    }

    getAddresses() {
        return axios.get(`${API_BASE_URL}/addresses`);
    }

    updateAddress(addressId, address) {
        return axios.put(`${API_BASE_URL}/address/${addressId}`, address);
    }

    deleteAddress(addressId) {
        return axios.delete(`${API_BASE_URL}/address/${addressId}`);
    }
}

export default new AddressService();
